mnefun/_mnefun.py

```diff
@@ -1411,7 +1419,22 @@ def save_epochs(p, subjects, in_names, in_numbers, analyses, out_names,
                 warnings.warn('resulting new sampling frequency %s not equal '
                               'to previous values %s' % (new_sfreq, sfreqs))
             sfreqs.add(new_sfreq)
-        use_reject, use_flat = _restrict_reject_flat(p.reject, p.flat, raw)
+        if p.autoreject_thresholds:
+            from autoreject import get_rejection_threshold
+            print('     Using autreject to compute rejection thresholds')
+            temp_epochs = Epochs(raw, events, event_id=None, tmin=p.tmin,
+                                 tmax=p.tmax, baseline=_get_baseline(p),
+                                 proj=True, reject=None, flat=None,
+                                 preload=True, decim=decim[si])
+            new_dict = get_rejection_threshold(temp_epochs)
+            use_reject, use_flat = _restrict_reject_flat(new_dict, p.flat, raw)
+            new_dict = get_rejection_threshold(temp_epochs)
+            use_reject = {k: new_dict[k]
+                          for k in ['grad', 'mag', 'eeg']
```

Comment:
Should this also have `eog` and `ecg` in it? I think our `reject` can usually have at least `eog`, but I'm not sure if `autoreject` does anything with the non-M/EEG channels

Reply:
So what happens if I use the entire rejection dictionary from autoreject e.g. `{'ecg': 2.93e-12, 'eog': 9.1e-05, 'eeg': 9.1e-05, 'grad': 7.97e-11, 'mag': 2.93e-12}` all epochs are rejected b/c ExG channels always exceed the autoreject criteria. In the case funloc subj_01 ECG063 causes all trials to be dropped. I assume this is because of how autoreject deals with ExG channels.

Reply:
This sounds like a bug with `autoreject`. Can you replicate the problem with the MNE sample dataset?

Reply:
I can run the question by Mainak, and maybe he'll look into it for us. But the other reason I opted for `['grad', 'mag', 'eeg']` is that IIRC in the past you have argued for using PCA to project out ExG artifacts rather than rejecting epochs based on their amplitudes.

Reply:
We'll have to talk to the `autoreject` folks about how these thresholds are chosen. We do want to keep the artifacts of interest, but not e.g. gigantic channel discontinuity artifacts. It's possible that even using `eeg` here would try to omit most of the blink epochs, for example, depending on how `autoreject` works.